store/cache.go

```diff
@@ -143,11 +147,46 @@ func (cm *CacheManager) cleanup() error {
 		return nil
 	}
 
-	sort.Sort(changesByReverseMtime(fil))
-	for _, fi := range fil[cm.maxItems:] {
-		if err := os.Remove(cm.path(fi.Name())); err != nil && os.IsNotExist(err) {
-			return err
+	var lastErr error
+	sort.Sort(changesByMtime(fil))
+	deleted := 0
+	for _, fi := range fil {
+		path := cm.path(fi.Name())
+		n, err := hardLinkCount(path)
```

Comment:
why pass `path` to `hardLinkCount` for it to stat, when you have the stat already in `fi`?

Reply:
Nice catch!